### PR TITLE
Add some transform apis

### PR DESCRIPTION
**Pull request by LielinJiang**
### PR types

New features
### PR changes

APIs
### Describe

Add some transform apis such as RandomCrop, RandomErasing, Pad, RandomRotate, Grayscale

image
image
image
image
image
image
image
image

### CODE REVIEW

File: python/paddle/incubate/hapi/vision/transforms/functional.py [hunk 1]
```diff
@@ -99,3 +101,163 @@ def resize(img, size, interpolation=cv2.INTER_LINEAR):
...
+            edge: pads with the last value on the edge of the image
+            reflect: pads with reflection of image (without repeating the last value on the edge)
+                padding [1, 2, 3, 4] with 2 elements on both sides in reflect mode
+                will result in [3, 2, 1, 2, 3, 4, 3, 2]
+            symmetric: pads with reflection of image (repeating the last value on the edge)
+                padding [1, 2, 3, 4] with 2 elements on both sides in symmetric mode
+                will result in [2, 1, 1, 2, 3, 4, 4, 3]
+
+    Returns:
+        numpy ndarray: Paded image.
```
Comment (heavengate) [inline python/paddle/incubate/hapi/vision/transforms/functional.py]: Paded -> Padded

File: python/paddle/incubate/hapi/vision/transforms/transforms.py [hunk 2]
```diff
@@ -800,3 +801,334 @@ def __init__(self, brightness=0, contrast=0, saturation=0, hue=0):
...
+class RandomCrop(object):
+    """Crop the given CV Image at a random location.
+    Args:
+        size (sequence|int): Desired output size of the crop. If size is an
+            int instead of sequence like (h, w), a square crop (size, size) is
+            made.
+        padding (int|sequence, optional): Optional padding on each border
+            of the image. Default is 0, i.e no padding. If a sequence of length
+            4 is provided, it is used to pad left, top, right, bottom borders
+            respectively.
```
Comment (heavengate) [inline python/paddle/incubate/hapi/vision/transforms/transforms.py]: add default value description?

File: python/paddle/incubate/hapi/vision/transforms/functional.py (hunk 1 above)
Comment (LielinJiang) [reply python/paddle/incubate/hapi/vision/transforms/functional.py]: done, thanks~

File: python/paddle/incubate/hapi/vision/transforms/transforms.py (hunk 2 above)
Comment (LielinJiang) [reply python/paddle/incubate/hapi/vision/transforms/transforms.py]: done, thanks

File: python/paddle/incubate/hapi/vision/transforms/functional.py [hunk 3]
```diff
@@ -99,3 +101,202 @@ def resize(img, size, interpolation=cv2.INTER_LINEAR):
...
+        padding (int|tuple): Padding on each border. If a single int is provided this
+            is used to pad all borders. If tuple of length 2 is provided this is the padding
+            on left/right and top/bottom respectively. If a tuple of length 4 is provided
+            this is the padding for the left, top, right and bottom borders
+            respectively.
+        fill (int|tuple): Pixel fill value for constant fill. Default is 0. If a tuple of
+            length 3, it is used to fill R, G, B channels respectively.
+            This value is only used when the padding_mode is constant
+        padding_mode: Type of padding. Should be: constant, edge, reflect or symmetric. Default is constant.
+            ``constant`` means pads with a constant value, this value is specified with fill.
```
Comment (Heeenrrry) [inline python/paddle/incubate/hapi/vision/transforms/functional.py]: "means padding"

File: python/paddle/incubate/hapi/vision/transforms/functional.py [hunk 4]
```diff
@@ -99,3 +101,202 @@ def resize(img, size, interpolation=cv2.INTER_LINEAR):
             return cv2.resize(img, (ow, oh), interpolation=interpolation)
     else:
         return cv2.resize(img, size[::-1], interpolation=interpolation)
+
+
+def pad(img, padding, fill=(0, 0, 0), padding_mode='constant'):
+    """Pad the given CV Image on all sides with speficified padding mode and fill value.
```
Comment (Heeenrrry) [inline python/paddle/incubate/hapi/vision/transforms/functional.py]: 应该用单数：Pad the given -> Pads the given

File: python/paddle/incubate/hapi/vision/transforms/functional.py [hunk 5]
```diff
@@ -99,3 +101,202 @@ def resize(img, size, interpolation=cv2.INTER_LINEAR):
...
+            respectively.
+        fill (int|tuple): Pixel fill value for constant fill. Default is 0. If a tuple of
+            length 3, it is used to fill R, G, B channels respectively.
+            This value is only used when the padding_mode is constant
+        padding_mode: Type of padding. Should be: constant, edge, reflect or symmetric. Default is constant.
+            ``constant`` means pads with a constant value, this value is specified with fill. 
+            ``edge`` means pads with the last value at the edge of the image. 
+            ``reflect`` means pads with reflection of image (without repeating the last value on the edge) 
+            padding ``[1, 2, 3, 4]`` with 2 elements on both sides in reflect mode 
+            will result in ``[3, 2, 1, 2, 3, 4, 3, 2]``
```
Comment (Heeenrrry) [inline python/paddle/incubate/hapi/vision/transforms/functional.py]: 句尾加句点

**Review by qingqing01 — COMMENTED** [review]
LGTM

**Review by Heeenrrry — APPROVED** [review]
LGTM

**Review by heavengate — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**